enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,241 @@
+	MaxDeviceCount int `json:"maxDeviceCount"`
+	// Volume mode. Raw or with filesystem
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// File system type to be used when volumeMode is filesystem
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// DeviceInclusionSpec are the filtration rules for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational,nonRotational and
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "nonRotational"
+	// RotationalAndNonRotational refers to both magnetic and ssds
+	RotationalAndNonRotational DeviceMechanicalProperty = "rotationalAndNonRotational"
+)
+
+type DeviceInclusionSpec struct {
+	// Device types that should be used for auto detection. This would be one of the types supported
+	// by the local-storage operator. Initial possible types can be - crypt, raid1, raid4, raid5,
+	// raid10, multipath, disk, tape, printer, processor, worm, rom, scanner, mo-disk, changer,
+	// comm, raid, enclosure, rbc, osd, and no-lun.
+	// If the list is empty the default deviceType will be Disk.
+	DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+	// The devices of this mechanicalPropery will be included.
+	// By default it is RotationalAndNonRotational(or SSD/HDD both)
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// The minimum size of the device which needs to be included
+	// +optional
+	MinSize resource.Quantity `json:"minSize"`
+
+	// The maximum size of the device which needs to be included
+	// +optional
+	MaxSize resource.Quantity `json:"maxSize"`
+
+	// List of models that needs to be included
+	// +optional
+	Models []string `json:"models"`
+
+	// List of vendors that needs to be
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+```
+
+An example of autoDetectVolume CR instance:
+```
+apiVersion: local.storage.openshift.io/v1
+kind: AutoDetectVolume
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector: null
+  deviceDiscoverPolicies:
+    - fsType: ext4
+      storageClassName: example-storageclass1
+      volumeMode: filesystem
+      minDeviceCount: 5
+      deviceInclusionSpec:
+        deviceType: part
+        deviceMechanicalProperty: nonRotational
+        minSize: 10G
+        maxSize: 100G
+    - storageClassName: example-storageclass2
+      volumeMode: Block
+      minDeviceCount: 5
+      maxDeviceCount: 10
+      deviceInclusionSpec:
+        deviceType: disk
+        deviceMechanicalProperty: rotational
+        minSize: 10G
+        maxSize: 100G
```

Comment:
Please add "model" and "vendor" property too. Please provide 2 examples: a simple one and a more advanced one. Thanks!

Reply:
add some more examples